applications/IgaApplication/custom_conditions/support_penalty_condition.cpp

```diff
@@ -0,0 +1,149 @@
+
+        // Memory allocation
+        if (CalculateStiffnessMatrixFlag) {
+            if (rLeftHandSideMatrix.size1() != mat_size) {
+                rLeftHandSideMatrix.resize(mat_size, mat_size, false);
+            }
+            noalias(rLeftHandSideMatrix) = ZeroMatrix(mat_size, mat_size);
+        }
+        if (CalculateResidualVectorFlag) {
+            if (rRightHandSideVector.size() != mat_size) {
+                rRightHandSideVector.resize(mat_size, false);
+            }
+            rRightHandSideVector = ZeroVector(mat_size);
+        }
+
+
+        if (Has(DISPLACEMENT))
+        {
+            // Integration
+            const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+            for (IndexType point_number = 0; point_number < integration_points.size(); point_number++)
+            {
+                Matrix N = r_geometry.ShapeFunctionsValues();
+
+                //FOR DISPLACEMENTS
+                Matrix H = ZeroMatrix(3, mat_size);
+                for (IndexType i = 0; i < number_of_nodes; i++)
+                {
+                    IndexType index = 3 * i;
+                    if (Is(IgaFlags::FIX_DISPLACEMENT_X))
+                        H(0, index) = N(point_number, i);
+                    if (Is(IgaFlags::FIX_DISPLACEMENT_Y))
+                        H(1, index + 1) = N(point_number, i);
+                    if (Is(IgaFlags::FIX_DISPLACEMENT_Z))
+                        H(2, index + 2) = N(point_number, i);
+                }
+
+                // Assembly
+                if (CalculateStiffnessMatrixFlag) {
+                    noalias(rLeftHandSideMatrix) += prod(trans(H), H)
+                        * penalty;
+                }
+                if (CalculateResidualVectorFlag) {
+
+                    const array_1d<double, 3>& displacement = this->GetValue(DISPLACEMENT);
+
+                    Vector u(mat_size);
+                    for (IndexType i = 0; i < number_of_nodes; i++)
+                    {
+                        const array_1d<double, 3> disp = r_geometry[i].FastGetSolutionStepValue(DISPLACEMENT);
+                        IndexType index = 3 * i;
+                        u[index]     = (disp[0] - displacement[0]);
+                        u[index + 1] = (disp[0] - displacement[0]);
+                        u[index + 2] = (disp[0] - displacement[0]);
+                    }
+
+                    noalias(rRightHandSideVector) -= prod(prod(trans(H), H), u)
+                        * penalty;
+                }
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+    int SupportPenaltyCondition::Check(const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(PENALTY_FACTOR)) << "No penalty factor (PENALTY_FACTOR) defined in property of SupportPenaltyPointDiscreteCondition" << std::endl;
+        return 0;
```

Comment:
Can you check if GetGeometry provides you a CurveOnSurfaceQuadraturePointGeometry.